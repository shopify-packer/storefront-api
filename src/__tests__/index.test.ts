import {StorefrontAPI} from "../index";
import {jest} from '@jest/globals';
import 'whatwg-fetch';
import {graphqlResponse, graphqlId} from './fixtures/response'
import {validConfig, invalidConfig} from "./fixtures/config";

beforeEach(async () => {
  jest.spyOn(global, "fetch").mockImplementation((req: Request) => {
    if (req.headers.get('x-shopify-storefront-access-token').length < 1) {
      return Promise.reject('Missing storefront access token')
    }
    const data = new Response(JSON.stringify(graphqlResponse))
    return Promise.resolve(data)
  })
})

test('Fetch query returns data', async () => {
  const storefrontAPI = new StorefrontAPI(validConfig)
  const data = await storefrontAPI.fetch('')
  const json = await data.json()
  expect(global.fetch).toHaveBeenCalledTimes(1)
  expect(json).toMatchSnapshot();
});

test('Decode all encoded shopify ids from a fetch response', async () => {
  const storefrontAPI = new StorefrontAPI(validConfig)
  const data = await storefrontAPI.fetch('')
  const json = await data.json()
  const decoded = storefrontAPI.decodeIds(json.data.collections)
  expect(decoded).toMatchSnapshot();
});

test('Invalid config returns error',  async () => {
  const storefrontAPI = new StorefrontAPI(invalidConfig)
  await expect(storefrontAPI.fetch('')).rejects.toMatch('Missing storefront access token')
});

test('Decode Shopify graphql id',  () => {
  const storefrontAPI = new StorefrontAPI(validConfig)
  expect(storefrontAPI.decodeId(graphqlId)).toMatchSnapshot()
});

