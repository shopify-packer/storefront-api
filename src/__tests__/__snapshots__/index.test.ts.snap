// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Decode Shopify graphql id 1`] = `"243638019"`;

exports[`Decode all encoded shopify ids from a fetch response 1`] = `
Object {
  "edges": Array [
    Object {
      "node": Object {
        "id": "243638019",
        "products": Object {
          "edges": Array [],
        },
        "title": "Home page",
      },
    },
    Object {
      "node": Object {
        "id": "257697667",
        "products": Object {
          "edges": Array [],
        },
        "title": "Antiperistaltic Gold Socks",
      },
    },
    Object {
      "node": Object {
        "id": "257697731",
        "products": Object {
          "edges": Array [],
        },
        "title": "Blistered Aluminum Boat",
      },
    },
    Object {
      "node": Object {
        "id": "389242179",
        "products": Object {
          "edges": Array [
            Object {
              "node": Object {
                "id": "9895309699",
                "title": "DKNYPURE PONCHO TOP",
              },
            },
            Object {
              "node": Object {
                "id": "9895307139",
                "title": "DKNYPURE TEE WITH UNDERLAYER",
              },
            },
          ],
        },
        "title": "Latest Stuff",
      },
    },
  ],
}
`;

exports[`Fetch query returns data 1`] = `
Object {
  "data": Object {
    "collections": Object {
      "edges": Array [
        Object {
          "node": Object {
            "id": "Z2lkOi8vc2hvcGlmeS9Db2xsZWN0aW9uLzI0MzYzODAxOQ==",
            "products": Object {
              "edges": Array [],
            },
            "title": "Home page",
          },
        },
        Object {
          "node": Object {
            "id": "Z2lkOi8vc2hvcGlmeS9Db2xsZWN0aW9uLzI1NzY5NzY2Nw==",
            "products": Object {
              "edges": Array [],
            },
            "title": "Antiperistaltic Gold Socks",
          },
        },
        Object {
          "node": Object {
            "id": "Z2lkOi8vc2hvcGlmeS9Db2xsZWN0aW9uLzI1NzY5NzczMQ==",
            "products": Object {
              "edges": Array [],
            },
            "title": "Blistered Aluminum Boat",
          },
        },
        Object {
          "node": Object {
            "id": "Z2lkOi8vc2hvcGlmeS9Db2xsZWN0aW9uLzM4OTI0MjE3OQ==",
            "products": Object {
              "edges": Array [
                Object {
                  "node": Object {
                    "id": "Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0Lzk4OTUzMDk2OTk=",
                    "title": "DKNYPURE PONCHO TOP",
                  },
                },
                Object {
                  "node": Object {
                    "id": "Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0Lzk4OTUzMDcxMzk=",
                    "title": "DKNYPURE TEE WITH UNDERLAYER",
                  },
                },
              ],
            },
            "title": "Latest Stuff",
          },
        },
      ],
    },
  },
}
`;
